examples/text-generation/textgen_pipeline/pipeline.py

```diff
@@ -0,0 +1,56 @@
+import os
+import sys
+
+import torch
+from transformers import TextGenerationPipeline
+
+
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.pardir)))
+try:
+    from utils import initialize_model
+except:
+    raise
+
+
+class GaudiTextGenerationPipeline(TextGenerationPipeline):
+    def __init__(self, args, logger):
+        self.model, self.tokenizer, self.generation_config = initialize_model(args, logger)
+
+        self.device = args.device
+
+        if args.do_sample:
+            self.generation_config.temperature = args.temperature
+            self.generation_config.top_p = args.top_p
+
+        self.max_padding_length = args.max_input_tokens if args.max_input_tokens > 0 else 100
+        self.use_hpu_graphs = args.use_hpu_graphs
+        self.profiling_steps = args.profiling_steps
+        self.profiling_warmup_steps = args.profiling_warmup_steps
+
+        import habana_frameworks.torch.hpu as torch_hpu
+
+        logger.info("Graph compilation...")
+        for _ in range(3):
+            self("Here is my prompt")
+        torch_hpu.synchronize()
+
+    def __call__(self, prompt: str):
+        model_inputs = self.tokenizer.encode_plus(
+            prompt, return_tensors="pt", max_length=self.max_padding_length, padding="max_length", truncation=True
+        )
```

Comment:
So we truncate sentences to `self.max_padding_length` if they are longer than `self.max_padding_length`?
and input sentences are always padded to `self.max_padding_length` ?

Reply:
Yes, that's correct. I believe this holds true for input tokens instead of input sentences.

Reply:
yes, right. by "input sentences" i meant "input sentences after being tuned into tokens"